Overhaul `@astrojs/sitemap` docs (#8997)
withastro/docs: src/content/docs/en/guides/integrations-guide/sitemap.mdx

`sitemap-index.xml` links to all the numbered sitemap files.
`sitemap-0.xml` lists the pages on your site.
For extremely large sites, there may also be additional numbered files like `sitemap-1.xml` and `sitemap-2.xml`.

<details>
<summary>Example of generated files for a two-page website</summary>

```xml title="sitemap-index.xml"
<?xml version="1.0" encoding="UTF-8"?>
  <sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
  <sitemap>
    <loc>https://stargazers.club/sitemap-0.xml</loc>
  </sitemap>
</sitemapindex>
```

```xml title="sitemap-0.xml"
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:news="http://www.google.com/schemas/sitemap-news/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
  <url>
    <loc>https://stargazers.club/</loc>
  </url>
  <url>
    <loc>https://stargazers.club/second-page/</loc>
  </url>
</urlset>
```
</details>

### Sitemap discovery

> You can make it easier for crawlers to find your sitemap with links in your site's `<head>` and `robots.txt` file.

Reviewer: ```suggestion
You can make it easier for crawlers to find your sitemap with links in your site's `<head>` or `robots.txt` file.
```
Just checking, would someone do *both* or only one of the following methods? If they're not supposed to add to both places (a conflict?) then I'd not use "and" this way. Could be swapping to "or" or could be extra context like:

> You can make it easier for crawlers to find your sitemap by including a link in either...

> To help crawlers discover your sitemap, you can add a link in one of the following locations...

Author: Yes, doing both is not uncommon. Do you still think “or” is better in that case?

Reviewer: Ah, nope! If doing both is common, then it's fine as is!

Sitemap: https://<YOUR SITE>/sitemap-index.xml
```

> You can also generate `robots.txt` dynamically if you want to reuse the `site` value from `astro.config.mjs`.
> To do so, create a `src/pages/robots.txt.ts` file with the following code:

Reviewer: ```suggestion
To generate a `robots.txt` file dynamically using the `site` value from `astro.config.mjs`, create the file `src/pages/robots.txt.ts` and add the following code:
```

Author: This removes the explanation for _why_ you might do this and almost makes it sound like you should use this dynamic approach as well as the static approach — which you shouldn’t. Maybe that should be clearer?

```suggestion
If you want to reuse the `site` value from `astro.config.mjs`, you can also generate `robots.txt` dynamically.
Instead of using a static file in the `public/` directory, create a `src/pages/robots.txt.ts` file and add the following code:
```

Reviewer: Yeah, I was worried about whether the why was clear enough in my suggestion. I didn't think it made it sound like it would do both, but if my suggestion also introduces that, then I am happy to withdraw my suggestion.

Mostly, I was trying to avoid "to do so" because that feels like we really should be able to write around that. :smile:
